### ISSUE TITLE
Save (update) User

### ISSUE THREAD

**Issue by joneslloyd** [WP-API/client-js#95]
Hi there,

Just wondering, is the User Save method available? I'm guessing not(?).

Is there any guidance on how I can add this in myself if it's not created?

Thanks!

**Comment by joneslloyd** [comment]
It may be that the issue is with trying to save when using the "Me" model..

I manually added "PUT" to the methods available for that model, and it tried to call:
`http://mysite.com/wp-json/wp/v2/users/me/7 `

(ie, appending the user id to the 'me' endpoint)

**Comment by joneslloyd** [comment]
Just to totally confirm, I think the issue is related to the above.

When I log the methods of the `wp.api.models.User` model to the console, I get:
`["GET", "POST", "PUT", "PATCH", "DELETE"]`

But when I do the same for `wp.api.models.Me` I get:
`["GET"]`

**Comment by adamsilverstein** [comment]
@joneslloyd 

Thanks for the bug report. The Me endpoint is indeed a special case; I will look into this more. Do you have a code snippet you can share so I can see exactly what you are trying to do? Thanks.

**Comment by joneslloyd** [comment]
No problem!

I don't have a particular snippet of code, rather, I'm trying to perform general actions..

For example, my set up is that the user logs in, and from then, they interact with a single page app in Backbone JS using WP-API.

They're able to update their first name / last name / email address / website, but currently I'm having to Extend the WP-API User (Me) model with my own, override the Save function and instantiate a normal user with the ID of the "me" user, save it, and then call the prototype of the User (Me) model afterwards.. Which is a bit hacky.

So basically I'm just looking to leverage the 'save' freedoms that the normal User model has.

Thanks!

**Comment by adamsilverstein** [comment]
@joneslloyd thanks for explaining. 

I'll take a look at the Me model to see if there is anything I can do to improve that experience. I agree it should behave more like a regular user model,part of the issue is that Me is a special case in the API currently. 

Have you tried this (still an extra step, but maybe cleaner): get the Me model, then create a new user model based on the me model's attributes. The new user model save() would then work as expected?

**Comment by adamsilverstein** [comment]
@joneslloyd I tested this a bit more. The underlying issue is that the `Me` endpoint only supports `GET` so the client correctly doesn't allow saving to this endpoint. I think this is expected behavior :( 

I found this approach worked: fetch the Me endpoint, create a User model from the returned model, edit and save that model. something like this:

```js
var me = new wp.api.models.Me();
me.fetch().done(... /* continue when loaded */
   var meUser = new wp.api.models.User( me.attributes );
       .... manipulate meUser
   meUser.save() /* works */
```

I would consider a PR to change the Me model and have it act more like a User model; for now this seems like a reasonable workaround.

Here is the endpoint details for `Me`
![endpoint](http://cl.ly/0E2q070c1e3j/wpdev.localhostwp-jsonwpv2_2016-02-25_09-41-27.jpg)
